src/common.js

```diff
@@ -119,7 +119,11 @@ function error(msg, _code, options) {
   state.error += logEntry;
 
   // Throw an error, or log the entry
-  if (options.fatal) throw new Error(logEntry);
+  if (options.fatal) {
+    var err = new Error(logEntry);
+    err.code = options.code || DEFAULT_OPTIONS.code;
```

Comment:
You can remove `|| DEFAULT_OPTIONS.code`. This is redundant because `options = Object.assign({}, DEFAULT_OPTIONS, options);` would already assign the value to `options.code`.